### PR TITLE
Added recursion documentation with examples : Issue no #113

### LINKED ISSUE #113: [Feature Request] Adding recursion documentation with examples.
### Is your feature request related to a problem? Please describe.
I'm frustrated when trying to understand complex recursion problems without proper documentation and clear examples. Currently, the project lacks comprehensive recursion-related explanations and example walkthroughs, making it difficult for users to grasp these concepts.

### Describe the solution you'd like
I would like to see detailed documentation on recursion, including common recursion problems such as:

- Rat in a Maze: Step-by-step explanation on how to navigate a maze using recursion.
- N Queen Problem: Demonstration of how recursion is used to place N queens on an N×N chessboard such that no two queens threaten each other.
- Sudoku Solver: A recursive approach to solving a given incomplete Sudoku board.

**Each problem should include:**
- Explanation of the problem.
- Pseudocode and/or actual code examples.
- Walkthrough of the recursion steps to enhance understanding.

### Describe alternatives you've considered
Alternatives could include embedding links to external resources, videos, or tutorials, but having everything consolidated in the project documentation would make it easier for users to refer to while coding.

### Additional context
This feature will improve the learning experience for users who are new to recursion or struggling with understanding recursion through practical examples. It could also serve as a useful resource for interview preparations or competitive programming.

---

**Would you like to work on this feature?**  
- [x] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by AbhijitMotekar99**
### Description
Added recursion examples and solutions for N-Queen, Sudoku, and Rat in a Maze problems in C++.

Fixes #113 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Recursion/recursion.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+# Recursion
```
Comment (ajay-dhangar) [inline docs/Recursion/recursion.md:10]: We need on top

```md
---
id: ___
title: ___
sidebar_label: ___
sidebar_position: ___
description: ___
tags: [___, __, ]
---
```

### Detailed Guidelines

- id: Unique identifier for the route.
- title: Title of the problem.
- sidebar_label: Problem number or label for the sidebar.
- tags: Relevant tags for the problem.
- description: A short SEO-friendly description. No HTML tags.

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
We need only `.md` file so, delete `.cpp` files. and update code in .md file based on suggestions

**Comment by AbhijitMotekar99** [comment]
Ok I will do it.

**Comment by AbhijitMotekar99** [comment]
Hey @ajay-dhangar 
I done all changes now you can merge.

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**